cmd/primarycmd/addValidator.go

```diff
@@ -192,10 +187,17 @@ func addValidator(_ *cobra.Command, _ []string) error {
 		return fmt.Errorf("illegal weight, must be greater than or equal to %d: %d", minValStake, weight)
 	}
 
-	kc, err := subnetcmd.GetKeychain(false, useLedger, ledgerAddresses, keyName, network)
+	fee := network.GenesisParams().AddPrimaryNetworkValidatorFee
+	kc, err := keychain.GetKeychain(app, false, useLedger, ledgerAddresses, keyName, network, fee)
 	if err != nil {
 		return err
 	}
+
+	// used in E2E to simulate public network execution paths on a local network
```

Comment:
nit: move this code to utils as we use it in many places

Reply:
will do it in separate PR to better manage and encapsulate this is the proper way. this needs more that a function, this needs to be called in a specific order and already want to encapsulate all in a bigger function